lib/core/Canvas.js

```diff
@@ -57,11 +58,13 @@ function createContainer(options) {
   return parent;
 }
 
-function createGroup(parent, cls) {
+function createGroup(parent, cls, childIndex) {
   var group = svgCreate('g');
   svgClasses(group).add(cls);
 
-  svgAppend(parent, group);
```

Comment:
Can it happen that no child nodes exist yet? In this case `parent.insertBefore(group, parent.childNodes[index]);` won't work.

Reply:
`childNodes` property always exists.

Reply:
Yea but what about `childNodes[0]` when there are none?

Reply:
I'll check again later! 👍

Reply:
This is not an issue. You can even do this:

```javascript
var parent = document.createElement('div'),
    child = document.createElement('div');

try {
  
  // insert at index 42
  parent.insertBefore(child, parent.childNodes[42]);
  
  console.log('success');
} catch(err) {
  console.log(err);
}
```

Reply:
Haha, nice one. So if the "before" marker element does not exist it will simply append. Good news.